targets/TARGET_Realtek/TARGET_AMEBA/TARGET_RTL8195A/device/TOOLCHAIN_IAR/rtl8195a.icf

```diff
@@ -1,323 +1,189 @@
-/*###ICF### Section handled by ICF editor, don't touch! ****/
-/*-Editor annotation file-*/
-/* IcfEditorFile="$TOOLKIT_DIR$\config\ide\IcfEditor\cortex_v1_0.xml" */
-/*-Specials-*/
-//define symbol __ICFEDIT_intvec_start__ = 0x00000000;
-
-//include "main.icf";
+/*
+ * Copyright (c) 2013-2017 Realtek Semiconductor Corp.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
 /*-Memory Regions-*/
-define symbol __ICFEDIT_region_ROM_start__                  = 0x00000000;
-define symbol __ICFEDIT_region_ROM_end__                    = 0x000FFFFF;
-define symbol __ICFEDIT_region_TCM_start__                  = 0x1FFF0000;
-define symbol __ICFEDIT_region_TCM_end__                    = 0x1FFFFFFF;
-define symbol __ICFEDIT_region_ROM_USED_RAM_start__         = 0x10000000;
-define symbol __ICFEDIT_region_ROM_USED_RAM_end__           = 0x10005FFF;
-//define symbol __ICFEDIT_region_RECY_RAM_start__           = 0x10002090;
-//define symbol __ICFEDIT_region_RECY_RAM_end__             = 0x100037FF;
-if( !isdefinedsymbol( __ICFEDIT_region_BD_RAM_start__ ) ) {
-    define symbol __ICFEDIT_region_BD_RAM_start__           = 0x10007000;
-}
-if( !isdefinedsymbol( __ICFEDIT_region_BD_RAM_end__ ) ) {
-    define symbol __ICFEDIT_region_BD_RAM_end__             = 0x1006FFFF;
-}
-define symbol __ICFEDIT_region_SDRAM_RAM_start__            = 0x30000000;
-define symbol __ICFEDIT_region_SDRAM_RAM_end__              = 0x301FFFFF;
-
-/*-Sizes-*/
-define symbol __ICFEDIT_size_cstack__   = 0x1000;
-define symbol __ICFEDIT_size_heap__     = 0x19000;
-/**** End of ICF editor section. ###ICF###*/
-
+define symbol __SRAM_start__  = 0x10007000;
```

Comment:
wondering why also name change:
- it was BD_RAM , now SRAM
- it was SDRAM, now DRAM

Reply:
Not losing any space. Removed regions are for ROM and TCM. ROM region is there for linking  only, which can be replaced with a symbol file. TCM region can be directly accessed in user program. 

Well, I found it very confusing when I was working on RTL8195AM. The BD_RAM is actually a fast SRAM.  In GCC's case, it's named SRAM1 and SRAM2. I will check if I can unify naming for all compilers.

Reply:
👍  I am on the same page, was looking at the linker scripts earlier and was confused. If they get unified , would be +1